modules/manifold/src/utils/utils.js

```diff
@@ -388,6 +358,32 @@ export function zipObjects(arrays, joinField, rename) {
   });
 }
 
+export function product(collectionArr) {
+  let result = [];
+  function recur(collection) {
```

Comment:
does this have to be done in a recursive way. Maybe a simple loop does the job more efficiently.

Reply:
There's no space or time complexity difference between recursion and iterative approach.